### PR TITLE
Add support for racetime.gg race comparisons

### PR DESCRIPTION
**Pull request by hwangbro**
Requires https://github.com/racetimeGG/racetime-app/pull/172 to be merged before working.

Adds support for creating and updating real time race comparisons for racetime.gg. A lot of the implementation for this copies the SRL implementation.

### CODE REVIEW

File: Controller/RacetimeChannel.cs [hunk 1]
```diff
@@ -414,12 +506,31 @@ private void State_OnUndoSplit(object sender, EventArgs e)
...
             {
                 if (PersonalStatus != UserStatus.Racing)
-                    SendChannelMessage(".undone");
+                    Undone();
+            }
+
+            if (PersonalStatus == UserStatus.Racing)
+            {
+                var split = Model.CurrentState.CurrentSplit;
+                string cmd = "{ \"action\": \"split\", \"data\": { \"split\":\"" + split.Name + "\", \"time\": \"-\"} }";
```
Comment (deains) [inline Controller/RacetimeChannel.cs]: Presumably C# has a JSON encode function somewhere? `split.Name` could contain `"` or other characters that could mess up encoding. I see as well we're using this for chat messages which is rather janky so might be good to fix that too!

**Review by deains — COMMENTED** [review]
👀

File: Controller/RacetimeChannel.cs (hunk 1 above)
Comment (wooferzfg) [reply Controller/RacetimeChannel.cs]: LiveSplit has a `DynamicJsonObject` class: https://github.com/LiveSplit/LiveSplit/blob/b893917f51245bfc5dfbc890114a32de76cfa0c8/LiveSplit/LiveSplit.Core/Web/JSON.cs#L123

File: Controller/RacetimeChannel.cs [hunk 2]
```diff
@@ -315,6 +323,8 @@ private void UpdateRaceData(RaceMessage msg)
             if (msg.Race != null)
                 Race = msg.Race;
 
+            UpdateRaceComparisons(Race);
```
Comment (Killklli) [inline Controller/RacetimeChannel.cs]: As msg.Race can be null in cases and we're only setting race in those chases, we should probably move this within the if statement

**Review by Killklli — CHANGES_REQUESTED** [review]

**Pull request merged**